cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/monitor/RuntimeMonitor.java

```diff
@@ -177,6 +220,12 @@ protected void run() {
             break;
           }
 
+          // Its possible that topicsToRequest is partially populated. So make sure we have all
```

Comment:
This comment seems not valid. You are checking if the map is empty, meaning it shouldn't be partially populated, right?

Reply:
Since it is in a loop, we still need to check if the list is empty, to avoid populating it each time loop is executed.